pkg/blockbuilder/blockbuilder.go

```diff
@@ -280,30 +342,47 @@ func (b *BlockBuilder) consumePartitionSection(
 			"last_consumed_offset", lastConsumedOffset, "num_blocks", len(blockMetas))
 	}(time.Now())
 
+	b.kafkaClient.ForceMetadataRefresh()
```

Comment:
(nit) Could you add a code-comment, mentioning why the ForceMetadataRefresh is needed here